eZ/Publish/Core/Persistence/Legacy/User/Handler.php

```diff
@@ -279,6 +355,38 @@ public function deleteRole($roleId)
     }
 
     /**
+     * Publish the specified role draft.
+     *
+     * @param mixed $roleId
+     */
+    public function publishRoleDraft($roleId)
+    {
+        $roleDraft = $this->loadRoleDraft($roleId);
+
+        $this->roleGateway->publishRoleDraft($roleDraft->id);
```

Comment:
@lolautruche Actually, I'm doing the delete-and-publish, which is why it works. Only I'm doing it in the RoleService, not in the Handler. If it's the wrong place I can move it.